apps/web/src/views/FarmAuction/helpers.tsx

```diff
@@ -81,12 +86,24 @@ const getAuctionStatus = (
   return AuctionStatus.ToBeAnnounced
 }
 
-const getDateForBlock = async (currentBlock: number, block: number) => {
+const getDateForBlock = async (chainId: ChainId, currentBlock: number, block: number) => {
   // if block already happened we can get timestamp via .getBlock(block)
   if (currentBlock > block) {
     try {
-      const { timestamp } = await publicClient({ chainId: ChainId.BSC }).getBlock({ blockNumber: BigInt(block) })
+      const { timestamp } = await publicClient({ chainId }).getBlock({ blockNumber: BigInt(block) })
       return dayjs.unix(Number(timestamp)).toDate()
+    } catch (error) {
+      // BSC Prod default public node won't return data for old blocks
+      if (chainId === ChainId.BSC) {
+        const randomNode = sample(
+          PUBLIC_NODES[chainId].filter((node) => node !== process.env.NEXT_PUBLIC_NODE_PRODUCTION),
+        )
+        const chain = CHAINS.find((c) => c.id === chainId)
+        const publicFallbackClient = createPublicClient({ chain, transport: http(randomNode), ...CLIENT_CONFIG })
+        const { timestamp } = await publicFallbackClient.getBlock({ blockNumber: BigInt(block) })
+        return dayjs.unix(Number(timestamp)).toDate()
+      }
+      throw error
```

Comment:
hmm didn't realize we still query for historical data. BTW, isn't this farm auction feature deprecated

Reply:
it's ok to remove farm auction related logic

Reply:
will stop support it when universal farm ready, feel free to remove the deprecated codes